src/main/java/games/strategy/engine/framework/GameRunner.java

```diff
@@ -522,44 +515,12 @@ public static void joinGame(final GameDescription description, final Messengers
     }
     final Version engineVersionOfGameToJoin = new Version(description.getEngineVersion());
     String newClassPath = null;
-    final boolean sameVersion =
-        ClientContext.engineVersion().equals(engineVersionOfGameToJoin);
+    final boolean sameVersion = ClientContext.engineVersion().equals(engineVersionOfGameToJoin);
     if (!sameVersion) {
-      try {
-        newClassPath = findOldJar(engineVersionOfGameToJoin, false);
-      } catch (final Exception e) {
-        if (ClientFileSystemHelper.areWeOldExtraJar()) {
-          JOptionPane.showMessageDialog(parent,
-              "<html>Please run the default TripleA and try joining the online lobby for it instead. "
-                  + "<br>This TripleA engine is old and kept only for backwards compatibility and can only play with "
-                  + "people using the exact same version as this one. "
-                  + "<br><br>Host is using a different engine than you, and cannot find correct engine: "
-                  + engineVersionOfGameToJoin.toStringFull("_") + "</html>",
-              "Correct TripleA Engine Not Found", JOptionPane.WARNING_MESSAGE);
-        } else {
-          JOptionPane.showMessageDialog(parent,
-              "Host is using a different engine than you, and cannot find correct engine: "
-                  + engineVersionOfGameToJoin.toStringFull("_"),
-              "Correct TripleA Engine Not Found", JOptionPane.WARNING_MESSAGE);
-        }
-        return;
-      }
-      // ask user if we really want to do this?
-      final String messageString = "<html>This TripleA engine is version "
-          + ClientContext.engineVersion()
-          + " and you are trying to join a game made with version " + engineVersionOfGameToJoin.toString()
-          + "<br>However, this TripleA can only play with engines that are the exact same version as itself (x_x_x_x)."
-          + "<br><br>TripleA now comes with older engines included with it, and has found the engine used by the host. "
-          + "This is a new feature and is in 'beta' stage."
-          + "<br>It will attempt to run a new instance of TripleA using the older engine jar file, and this instance "
-          + "will join the host's game."
-          + "<br>Your current instance will not be closed. Please report any bugs or issues."
-          + "<br><br>Do you wish to continue?</html>";
-      final int answer = JOptionPane.showConfirmDialog(null, messageString, "Run old jar to join hosted game?",
-          JOptionPane.YES_NO_OPTION);
-      if (answer != JOptionPane.YES_OPTION) {
-        return;
-      }
+      JOptionPane.showMessageDialog(parent,
+          "Host is using version " + engineVersionOfGameToJoin.toStringFull("_")
+              + ". You need to have the same engine version in order to join this game.",
+          "Incompatible TripleA engine", JOptionPane.ERROR_MESSAGE);
     }
```

Comment:
Should there be a  `return` after displaying the message dialog so `joinGame()` is not executed below?

Reply:
Good catch!

Reply:
This should also fix the "duplicate error message" issue I described ealier